src/Test/Desktop/OpenRiaServices.Common.Test/ExceptionHelper.cs

```diff
@@ -81,6 +83,50 @@ private static TException ExpectExceptionHelper<TException>(GenericDelegate del,
             return ExpectException<TException>(del, false);
         }
 
+
+#if !SILVERLIGHT
+        private static Task ExecuteDelegate(AsyncDelegate asyncDelegate)
+        {
+            try
+            {
+                return asyncDelegate();
+            }
+            catch (Exception ex)
+            {
+                var tcs = new TaskCompletionSource<int>();
+                tcs.SetException(ex);
+                return tcs.Task;
+            }
+        }
+
+        public static Task<TException> ExpectException<TException>(AsyncDelegate asyncDel) where TException : Exception
+        {
+            return ExecuteDelegate(asyncDel)
+                .ContinueWith(res =>
+                {
+                    return ExpectException<TException>(() => res.GetAwaiter().GetResult(), false);
+                });
+        }
+
+        public static Task<TException> ExpectException<TException>(AsyncDelegate del, string exceptionMessage)
```

Comment:
Consider renaming to ExpectExceptionAsync so test code can be reviewd to see that it is safe